tests/test_driver.py

```diff
@@ -61,13 +50,10 @@ def test_is_closed(self):
             { "mocked_state": 1, "inverted": True, "expected_relay_state": False },
             { "mocked_state": 0, "inverted": True, "expected_relay_state": True },
         ]
-        for case in cases:
-            GPIO_mock.input = Mock(return_value=case["mocked_state"])
-            relay = Relay(18, case["inverted"])
+        for case_id, case in enumerate(cases):
+            relay = Relay.get_or_create_relay(case_id + 10, case["inverted"], MockFactory())
```

Comment:
consider keeping the `Relay()` (constructor) here to reduce this magic with `+10`.
That allows to keep each test case independent from cache, thus it does not need different pins and you could keep the original `18` for them all

Reply:
the more straightforward the test data is — it's better

Reply:
Yes, so that you can minimize the changes of your PR just by keeping the `Relay()` and testing that new static method separately.

Reply:
fixed